frame/curve-amm/src/maths.rs

```diff
@@ -61,7 +61,10 @@ pub fn compute_d(
 
 		// d = (ann * sum + d_p * n) * d / ((ann - 1) * d + (n + 1) * d_p)
 		let mut term1 = ann.clone().mul(&sum).add(&d_p.clone().mul(&n)).mul(&d);
-		let mut term2 = d.mul(&ann_one).add(&n.clone().add(&one).mul(&d_p));
+		let mut term2 = ann_d
+			.add(&n.clone().add(&one).mul(&d_p))
```

Comment:
where is -1 from formula?

Reply:
need to update comment.if `ann` is 0 `ann - 1` will underflow.